vortex-datafusion/src/persistent/format.rs

```diff
@@ -73,12 +75,29 @@ impl FileFormat for VortexFormat {
         objects: &[ObjectMeta],
     ) -> DFResult<SchemaRef> {
         let mut file_schemas = Vec::default();
-        for o in objects {
-            let os_read_at = ObjectStoreReadAt::new(store.clone(), o.location.clone());
-            let initial_read = read_initial_bytes(&os_read_at, o.size as u64).await?;
-            let lazy_dtype = initial_read.lazy_dtype();
-            let s = infer_schema(lazy_dtype.value()?)?;
-            file_schemas.push(s);
+        let mut tasks: JoinSet<DFResult<Schema>> = objects
+            .iter()
+            .map(|o| {
+                let store = store.clone();
+                let o = o.clone();
+                async move {
+                    let os_read_at = ObjectStoreReadAt::new(store, o.location.clone());
+                    let initial_read = read_initial_bytes(&os_read_at, o.size as u64).await?;
```

Comment:
Also do we just throw this initial read away? Don't we need it again? How many times do we call this?

Reply:
this code path should only happen once per session

Reply:
Hm, we do read it once again to read the layouts. Would be nice to be able to share them but no clue what datafusion lets you store across those calls

Reply:
Feels a bit implicit, let's figure out how to actually just re-use the initial read

Reply:
I think though we should merge this and figure out reuse independently